process/scToProtocol/stakingToPeer.go

```diff
@@ -193,12 +188,18 @@ func (stp *stakingToPeer) UpdateProtocol(body *block.Body, nonce uint64) error {
 
 func (stp *stakingToPeer) updatePeerState(
 	stakingData systemSmartContracts.StakedData,
-	account state.PeerAccountHandler,
 	blsPubKey []byte,
 	nonce uint64,
 ) error {
+	if stakingData.StakedNonce == math.MaxUint64 {
```

Comment:
ugly hardcoded thing. Will think about it.

Reply:
yes - will try to get rid of this component and make the stakingSC to write directly into the validator trie - but there are a few issues right now with it. Like like reverting and journalizing of those elements.